### DISCUSSION TITLE
FBDefines/FBDefines.h not found

### DISCUSSION THREAD

**Discussion by FedorSh-illumiti** [invertase/react-native-firebase#6446] [General]
Hi. After the version upgrade from ^14.11.1 to 15.2.0 the project cannot be built in Xcode. 

podfile

require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '15.0'

target 'FDC' do
  config = use_native_modules!

  use_react_native!(
    :path => config[:reactNativePath],
    # to enable hermes on iOS, change `false` to `true` and then install pods
    :hermes_enabled => false
  )

  use_frameworks! :linkage => :static

  target 'FDC' do
    inherit! :complete
    # Pods for testing
  end

  use_flipper!({ 'Flipper' => '0.156.0' })
  post_install do |installer|
    react_native_post_install(installer)
    __apply_Xcode_12_5_M1_post_install_workaround(installer)
  end
end

**Comment by davidepalazzo** [comment]
Hi there. That's because Flipper doesn't work when `use_frameworks!` is enabled. It's stated in the [RN basic template Podfile](https://github.com/facebook/react-native/blob/main/template/ios/Podfile#L22).

Try the setup mentioned in [Flipper's documentation](https://fbflipper.com/docs/getting-started/ios-native/) (check out line 27).

**Reply by FedorSh-illumiti** [reply]
Hi. Thanks for your response. I removed use_frameworks! :linkage => :static. I added "use_modular_headers!'. Now I am getting the error  "/Users/.../Documents/Projects/Production/btp-react-native-field-data-capture/ios/Pods/Headers/Public/glog/glog/logging.h:512:1: Import of module 'glog.glog.log_severity' appears within namespace 'google'".


modfied Podfile:

require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '15.0'


use_modular_headers!

target 'FDC' do
  config = use_native_modules!

  use_react_native!(
    :path => config[:reactNativePath],
    # to enable hermes on iOS, change `false` to `true` and then install pods
    :hermes_enabled => false
  )




  target 'FDCTests' do
    inherit! :complete
    # Pods for testing
  end


  use_flipper!({ 'Flipper' => '0.156.0' })
  post_install do |installer|
    react_native_post_install(installer)
    __apply_Xcode_12_5_M1_post_install_workaround(installer)
  end
end

**Reply by davidepalazzo** [reply]
Try disabling the modular import for that package:

```ruby
pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec', :modular_headers => false
```

**Reply by FedorSh-illumiti** [reply]
still error

Import of module ‘glog.glog.log_severity” appears within namespace ‘google’

Showing Recent Errors Only
/Users/../Documents/Projects/Production/FDC/node_modules/react-native/ReactCommon/logger/react_native_log.cpp:9:10: in file included from /Users/../Documents/Projects/Production/FDC/node_modules/react-native/ReactCommon/logger/react_native_log.cpp:9:

**Reply by davidepalazzo** [reply]
Does it build with Flipper disabled? A minimal reproducible repo would be good.

**Comment by Muhammad-Sufian** [comment]
https://stackoverflow.com/questions/73894220/flipperkit-flipperclient-h-file-not-found

this resolved the error for me
